server/src/main/java/org/elasticsearch/action/admin/indices/shards/TransportIndicesShardStoresAction.java

```diff
@@ -66,6 +63,8 @@ public class TransportIndicesShardStoresAction extends TransportMasterNodeReadAc
 
     private static final Logger logger = LogManager.getLogger(TransportIndicesShardStoresAction.class);
 
+    static final int CONCURRENT_REQUESTS_LIMIT = 100; // TODO configurable?
```

Comment:
Should it be converted to a Setting or was it intentionally left as a TODO?
Not sure if this is important though as it was not limited previously

Reply:
Well it's worth a discussion. Bounding concurrency here might make things slower. 100 feels high enough to me to have limited impact, whilst being low enough to avoid this API being actively harmful to an enormous cluster.

Reply:
Seems it could also depend on CPU/network. Have we ever recorded this was an issue in the wild when it was unlimited?
If not I am leaning towards approach of making it configurable with a huge default (1000?) and let them set it lower if/when  issue is detected.